sdk/core/core-lro/docs/MIGRATION.md

```diff
@@ -0,0 +1,94 @@
+# Migration Guide
+
+This document shows the customers of v2 LRO (Long Running Operations) on how to migrate their code to use the v3 libraries.
+
+## Main difference between v2 and v3
+
+In v2, you have to `await` the LRO to get this poller. But `await` doesn't give you the final result, it gives you the poller to track the operation's status. This has confused some users who thought `await` would give them the final result.
+
+To make things clearer and easier to understand, we've changed how this works in v3. Now, our poller interface blends the need to track long-running operations with the usual way of handling asynchronous programming. And this is the main difference between them.
+
+Except that the `LroEngine` is no longer supported and you could use `createHttpPoller` to create a poller instead.
+
+If you have an existing application that uses our v2 libaries and you're interested in updating your application to use the latest one, here are the things that you need to do for the migration:
+
+## Creation of Poller
+
+In v2 we have two implementations for LRO: `LroEngine` and `SimplePollerLike` with `createHttpPoller`. In v3 we deprecate the former and keep the `PollerLike` with `createHttpPoller` to build the poller. And the return type is changed from `SimplePollerLike` to `PollerLike`. The creation code would be changed as below.
+
+From
+
+```typescript
+const pollerSetting: LongRunningOperation<TResult> = {
+  requestMethod: "{http-method}",
+  requestPath: "{path}",
+  sendInitialRequest: async () => {
+    // your code to send initial request
+  },
+  sendPollRequest: async (path) => {
+    // your code to trigger polling request
+  },
+};
+const httpPoller = await createHttpPoller(pollerSetting, options);
+```
+
+To
+
+```typescript
+const pollerSetting: LongRunningOperation<TResult> = {
```

Comment:
The guide will need to use the updated names. Also, please ask the architects to take a look at them as well.